### PR TITLE
[CodingStandards] use short array syntax, unused elements, unused namespaces

### PR DESCRIPTION
**Pull request by TomasVotruba**
| Q               | A
| --------------- | ---
| Bug fix?        | no |
| New feature?    | no |
| BC breaks?      | no |
| License         | MIT |

This is my attempt to cleanup code a bit, since I saw PHP 7.1 on the move.

### CODE REVIEW

**Comment by TomasVotruba** [comment]
After removing some dependencies from constructor, some dependencies are missing: https://travis-ci.org/Sylius/Sylius/jobs/273473971#L1091

Why not use autowiring?
Where are they set?

I can fix that but first let me know, what changes are you open to.

File: src/Sylius/Bundle/CoreBundle/Form/Extension/ShippingMethodTypeExtension.php [hunk 1]
```diff
@@ -28,36 +26,6 @@
...
-    /**
-     * @param ServiceRegistryInterface $checkerRegistry
-     * @param ServiceRegistryInterface $calculatorRegistry
-     * @param FormTypeRegistryInterface $formTypeRegistry
-     */
-    public function __construct(
-        ServiceRegistryInterface $checkerRegistry,
-        ServiceRegistryInterface $calculatorRegistry,
-        FormTypeRegistryInterface $formTypeRegistry
-    ) {
```
Comment (pamil) [inline src/Sylius/Bundle/CoreBundle/Form/Extension/ShippingMethodTypeExtension.php]: Let's make sure we delete those from the service definition.

File: src/Sylius/Bundle/PayumBundle/Controller/PayumController.php [hunk 2]
```diff
@@ -84,15 +79,13 @@
...
         OrderRepositoryInterface $orderRepository,
-        FormTypeRegistryInterface $gatewayConfigurationTypeRegistry,
         MetadataInterface $orderMetadata,
         RequestConfigurationFactoryInterface $requestConfigurationFactory,
         ViewHandlerInterface $viewHandler,
         RouterInterface $router
     ) {
         $this->payum = $payum;
         $this->orderRepository = $orderRepository;
-        $this->gatewayConfigurationTypeRegistry = $gatewayConfigurationTypeRegistry;
```
Comment (pamil) [inline src/Sylius/Bundle/PayumBundle/Controller/PayumController.php]: Same here.

**Comment by pamil** [comment]
Btw. we can't introduce autowiring this late, so I wouldn't even consider it. Let's just customise these services definitions to match the constructors.

**Comment by TomasVotruba** [comment]
Sure, could you link me to them?

**Comment by TomasVotruba** [comment]
Ok, here is my attempt

**Comment by TomasVotruba** [comment]
Btw, when you decide to autowire, I recommend jumping right to [PSR-4 autodiscovery](https://www.tomasvotruba.cz/blog/2017/05/07/how-to-refactor-to-new-dependency-injection-features-in-symfony-3-3/#4-use-psr-4-based-service-autodiscovery-and-registration). Very addictive and easy to use. I never forgot to register service again :)

**Comment by pamil** [comment]
Thanks Tomas! 🎉

**Comment by pamil** [comment]
Btw. we should consider some of these checks to be a part of our CI process, so that PRs without sorted imports / with unused private properties won't get merged.

**Comment by TomasVotruba** [comment]
Exactly my point :) 
I used [Easy Coding Standard](https://github.com/Symplify/EasyCodingStandard) with 5 checkers for this.

**Comment by TomasVotruba** [comment]
You're welcome

File: src/Sylius/Bundle/CoreBundle/Form/Extension/ShippingMethodTypeExtension.php (hunk 1 above)
Comment (lchrusciel) [reply src/Sylius/Bundle/CoreBundle/Form/Extension/ShippingMethodTypeExtension.php]: I guess we should mention about these changes in upgrade file. Even if it was not used it is a BC at the end

**Comment by TomasVotruba** [comment]
I used this setup, to easier reproduce in the future:

### Install

```bash
composer create-project symplify/easy-coding-standard temp/ecs
```

### Setup

```yaml
# easy-coding-standard.neon
checkers:
    PhpCsFixer\Fixer\ArrayNotation\ArraySyntaxFixer:
        syntax: short
    - SlevomatCodingStandard\Sniffs\Classes\UnusedPrivateElementsSniff
    - PhpCsFixer\Fixer\Import\NoUnusedImportsFixer
    - PhpCsFixer\Fixer\Import\OrderedImportsFixer
```

### Check (dry only)

```bash
temp/ecs/bin/ecs check src tests 
```

### Fix

```bash
temp/ecs/bin/ecs check src tests  --fix
```

**Comment by TomasVotruba** [comment]
I do this every week with various open-source projects, so I put that to a simple post to easily replicate:
https://www.tomasvotruba.cz/blog/2017/09/15/4-simple-checkers-for-coding-standard-haters-but-clean-code-lovers/

**Pull request merged**